t/unit-tests/u-oidmap.c

```diff
@@ -0,0 +1,152 @@
+#include "unit-test.h"
+#include "lib-oid.h"
+#include "oidmap.h"
+#include "hash.h"
+#include "hex.h"
+
+/*
+ * Elements we will put in oidmap structs are made of a key: the entry.oid
+ * field, which is of type struct object_id, and a value: the name field (could
+ * be a refname for example).
+ */
+struct test_entry {
+	struct oidmap_entry entry;
+	char name[FLEX_ARRAY];
+};
+
+static const char *const key_val[][2] = { { "11", "one" },
+					  { "22", "two" },
+					  { "33", "three" } };
+
+static void setup(void (*f)(struct oidmap *map))
+{
+	struct oidmap map = OIDMAP_INIT;
+
+	for (size_t i = 0; i < ARRAY_SIZE(key_val); i++){
+		struct test_entry *entry;
+
+		FLEX_ALLOC_STR(entry, name, key_val[i][1]);
+		cl_parse_any_oid(key_val[i][0], &entry->entry.oid);
+		entry = oidmap_put(&map, entry);
+		cl_assert(entry == NULL);
```

Comment:
You can probably drop the `entry` variable and directly `cl_assert_equal_p(oidmap_put(&map, entry), NULL)`. The same is also true in other cases.